### ISSUE TITLE
Support visualizing categorical/ordinal indicators over time

### ISSUE THREAD

**Issue by paarriagadap** [owid/owid-grapher#4178]
## Core problem

We currently don't have a good way to handle categorical variables across time in chart views. We usually plot them as a map and without a chart (see [WB's income groups](https://ourworldindata.org/grapher/world-bank-income-groups), for example).

## Proposed solution

Create a new chart type to handle categorical variables. It could be a timeline-Gantt-like type of chart or anything that allows users to see how a country/entity moves between categories across time.

## Alternatives

There is currently an alternative approach, by recoding categories as numbers and show line charts, but this is suboptimal since the plots don't reference the categories and we are forced to use FASTT space to specify what the numbers mean. These numbers also don't correspond to the categories in the map view. See for example this [political regime chart](https://ourworldindata.org/grapher/political-regime).

## Context

Conversation in Slack [here](https://owid.slack.com/archives/C0149NKLZAM/p1731927955021769). Sophia mentions that we are currently in a much better position to handle this, since Grapher now knows the type of data we are dealing with.

**Comment by marcelgerber** [comment]
Plotting categorical (ordinal) data over time is not super easy, but can be done.

One option is what we are doing for map tooltips currently: Have a line chart where the y axis is labelled with the various category names, and ideally color these categories to make it easier to parse.

![Image](https://github.com/user-attachments/assets/e92cb33b-4034-40f2-a829-5c160165d2fa)

Another option is to have a "categorical scatter", where each observation is marked by a dot (again, in a y axis with category names).
![Image](https://github.com/user-attachments/assets/9afe4e4c-02ab-4f22-b791-262f31ca3e89)


One problem about both of these is that it is a bit tricky to extend these to multi-entity views without being confusing - so, potentially, this would be a single-entity chart type, then?

**Comment by paarriagadap** [comment]
Thanks @marcelgerber! Yeah, I also thought about the map tooltip. Maybe that could be an easier option to implement.

Regarding the use scatter plots, it is somewhat similar to what [Datawrapper recommends here](https://academy.datawrapper.de/article/292-how-to-create-a-timeline-using-scatter-plots).

And yeah, they could probably not be in multi-entity views, I was thinking on single-entity chart type or faceted ([like what Bastian did here](https://ourworldindata.org/grapher/political-regime?tab=chart)).

**Comment by sophiamersmann** [comment]
The drawback of these workarounds is that categorical data has to be mapped to numerical data, which makes the data table and chart tooltips difficult to read. The Graphers would be much easier to read/understand if that mapping wasn't necessary for a chart to be rendered.

**Comment by marcelgerber** [comment]
Yes! I was thinking of these screenshots as a starting point for an implementation improves on them.
I agree that we should get rid of the mapping to numbers, if at all possible!

**Comment by marcelgerber** [comment]
Saloni's current data insight is a nice example of such a categorical (here: ordinal) chart:

![Image](https://github.com/user-attachments/assets/9c353ce1-ce1d-466a-bb6b-51d6c1f7ffe9)

If the data is non-ordinal, then we could hide the connecting lines and plot dots instead (which we can now do 👑).
What's important, though, is that the y axis is showing the labels as they apply to this data.

**Comment by danyx23** [comment]
Closing this in favor of the project card
